internal/handler/web/users.go

```diff
@@ -89,3 +89,30 @@ func (h *Handler) PostUsersApprovedResearchersImportCsv(ctx *gin.Context) {
 	}
 	ctx.Status(http.StatusNoContent)
 }
+
+func (h *Handler) PostUsersInvite(ctx *gin.Context) {
+	user := middleware.GetUser(ctx)
+	attributes, err := h.users.Attributes(user)
+	if err != nil {
+		setError(ctx, err, "Failed to get user attributes")
+		return
+	}
+
+	sponsor := types.Sponsor{
+		Username:   user.Username,
+		ChosenName: attributes.ChosenName,
+	}
+
+	var invite openapi.PostUsersInviteJSONRequestBody
+	if err := bindJSONOrSetError(ctx, &invite); err != nil {
+		return
+	}
```

Comment:
nit: probably pointless optimisation (sorry!) – if the request is invalid then we make an unnecessary database call. I'd put the JSON bind at the top of the function